nuxeo-core/nuxeo-core-cache/src/main/java/org/nuxeo/ecm/core/transientstore/keyvalueblob/KeyValueBlobTransientStore.java

```diff
@@ -239,10 +239,8 @@ public void doGC() {
         BlobProvider bp = getBlobProvider();
         BinaryGarbageCollector gc = bp.getBinaryManager().getGarbageCollector();
         gc.start();
-        keyStream().map(this::getBlobs) //
+        keyStream().map(this::getBlobKeys) //
```

Comment:
Consider invoking gc.stop(true) in finally block to ensure gc is always stopped.
Also, handle potential concurrent blobKey removal between this.getBlobKeys() and gc.mark().

Reply:
Yes that's planned for an updated commit, this PR was not really for review yet.